plasmapy/utils/code_repr.py

```diff
@@ -0,0 +1,321 @@
+        string = " ".join(string.split())
+        string = string.replace(" ,", ",")
+        string = string.replace("[ ", "[")
+        return string
+
+    def put_np_before_infs_and_nans(string: str) -> str:
+        string = string.replace("inf", "np.inf")
+        string = string.replace("nan", "np.nan")
+        return string
+
+    def replace_excess_items_with_ellipsis(s, max_items):
+        substrings_between_commas = s.split(",")
+        to_comma_before_ellipsis = ",".join(substrings_between_commas[0:max_items])
+        closing_brackets = "]" * substrings_between_commas[-1].count("]")
+        closing = f", ...{closing_brackets})"
+        return to_comma_before_ellipsis + closing
+
+    if not isinstance(array, np.ndarray):
+        raise TypeError("Expecting an ndarray.")
+
+    s = np.array_repr(array, max_line_width=np.inf, suppress_small=False)
+    s = remove_excess_spaces(s)
+    s = put_np_before_infs_and_nans(s)
+
+    if array.size > max_items:
+        s = replace_excess_items_with_ellipsis(s, max_items)
+
+    return f"np.{s}"
+
+
+def _code_repr_of_quantity(arg: u.Quantity) -> str:
+    """
+    Transform a `~astropy.units.Quantity` into a format as would appear
+    in a function call.
+    """
+    if not isinstance(arg, u.Quantity):
+        raise TypeError("Expecting a Quantity.")
+
+    is_ndarray = isinstance(arg.value, np.ndarray)
+    formatted = _code_repr_of_ndarray(arg.value) if is_ndarray else repr(arg.value)
+
+    if arg.unit == u.dimensionless_unscaled:
+        formatted += "*u.dimensionless_unscaled"
+    else:
+        for base, power in zip(arg.unit.bases, arg.unit.powers):
+            if power == -1:
+                formatted += f"/u.{base}"
+            elif power == 1:
+                formatted += f"*u.{base}"
+            else:
+                formatted += f"*u.{base}**{power}"
+
+    return formatted
+
+
+def _code_repr_of_arg(arg) -> str:
+    """Transform an argument into a format as would appear in a function call."""
+    if hasattr(arg, "__name__"):
+        return arg.__name__
+
+    function_for_type = {
+        u.Quantity: _code_repr_of_quantity,
+        np.ndarray: _code_repr_of_ndarray,
+    }
+
+    for arg_type in function_for_type.keys():
+        if isinstance(arg, arg_type):
+            code_repr_func = function_for_type[arg_type]
+            return code_repr_func(arg)
+
+    return repr(arg)
+
+
+def _code_repr_of_args_and_kwargs(args: Any = tuple(), kwargs: Dict = {}) -> str:
+    """
+    Take positional and keyword arguments, and format them into a
+    string as they would appear in a function call (excluding parentheses).
+    """
+    args_collection = args if isinstance(args, tuple) else (args,)
+
+    args_and_kwargs = ""
+
+    for arg in args_collection:
+        args_and_kwargs += f"{_code_repr_of_arg(arg)}, "
+
+    for kwarg in kwargs.keys():
+        args_and_kwargs += f"{kwarg}={_code_repr_of_arg(kwargs[kwarg])}, "
```

Comment:
```suggestion
    for kw_name, kw_val in kwargs.items():
        args_and_kwargs += f"{kw_name}={_code_repr_of_arg(kw_val)}, "
```

Just think this is a little more readable...but your decision to take it or leave it.